resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/refresh/win32/Win32Natives.java

```diff
@@ -11,142 +11,145 @@
  * Contributors:
  *     IBM - Initial API and implementation
  *     Mikael Barbero (Eclipse Foundation) - 286681 handle WAIT_ABANDONED_0 return value
+ *     Hannes Wellmann - Migrate Win32Natives to use JNA instead of JNI to avoid native binaries and platform-specific fragments
  *******************************************************************************/
+
 package org.eclipse.core.internal.resources.refresh.win32;
 
+import com.sun.jna.Native;
+import com.sun.jna.NativeLibrary;
+import com.sun.jna.Pointer;
+import com.sun.jna.WString;
+import com.sun.jna.platform.win32.WinBase;
+import com.sun.jna.platform.win32.WinError;
+import com.sun.jna.platform.win32.WinNT;
+
 /**
  * Hooks for native methods involved with win32 auto-refresh callbacks.
  */
 public class Win32Natives {
+
+	/** Access is denied. */
+	public static final int ERROR_ACCESS_DENIED = 5;
+
+	// All constants and native methods are defined with the exact same name
+	// in the native windows.h or a referenced header file.
+
 	/* general purpose */
 	/**
 	 * A general use constant expressing the value of an
 	 * invalid handle.
 	 */
-	public static final long INVALID_HANDLE_VALUE;
+	public static final long INVALID_HANDLE_VALUE = Pointer.nativeValue(WinBase.INVALID_HANDLE_VALUE.getPointer());
```

Comment:
If the values of these constants is just inlined, we could even save the dependency to `jna-platform` and would just depend on `jna`.